core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -253,7 +253,7 @@ CudaInternal &CudaInternal::singleton() {
   return self;
 }
 void CudaInternal::fence(const std::string &name) const {
-  Impl::cuda_stream_synchronize(get_stream(), this, name);
+  Impl::cuda_stream_synchronize(m_stream, this, name);
```

Comment:
@romintomasetti @masterleinad The changes in https://github.com/kokkos/kokkos/pull/7569 did not remove all `cudaSetDevice` calls for fencing since this call to `get_stream` was also setting the device.

We could use `get_stream<false>()`, but I think we should remove the `CudaInternal::get_stream()` function as a part of a bigger clean up/rethink of the wrappers. But that is probably not related enough to this PR.

Reply:
I agree that you should remove the getter for the public member `m_stream`.